src/Command/SubProfileCommand.php

```diff
@@ -0,0 +1,351 @@
+        $question->setMultiselect(true);
+        $io->writeln('The following sub-components can be excluded: ' . Element::oxford($question->getChoices()));
+        $excluded_subcomponents = $io->askChoiceQuestion($question);
+      }
+      $input->setOption('excluded_subcomponents', $excluded_subcomponents);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    $io = new DrupalStyle($input, $output);
+
+    if (!$this->confirmGeneration($io)) {
+      return;
+    }
+
+    $profile = $this->validator->validateModuleName($input->getOption('profile'));
+    $machine_name = $this->validator->validateMachineName($input->getOption('machine-name'));
+    $description = $input->getOption('description');
+    $profile_path = $this->appRoot . '/profiles/custom';
+
+    // Check if all module dependencies are available.
+    $dependencies = $this->validator->validateModuleDependencies($input->getOption('dependencies'));
+    if ($dependencies) {
+      // @todo ProfileCommand::checkDependencies is private so we aren't
+      // actually checking to see if they are present.
+      $dependencies = $dependencies['success'];
+    }
+    $this->buildExcludedDependenciesList($input->getOption('excluded_dependencies'));
+    $this->buildExcludedDependenciesList($input->getOption('excluded_subcomponents'));
+    $this->generator->generate(
+      $profile,
+      $machine_name,
+      $profile_path,
+      $description,
+      $dependencies,
+      $this->excludedDependencies
+    );
+  }
+
+  /**
+   * @param array $excluded_dependencies
+   *   Dependencies to exclude.
+   *
+   * Adds the provided component(s) to the excluded list and, if the provided
+   * component is a top-level component, all of its subcomponents too.
+   */
+  protected function buildExcludedDependenciesList($excluded_dependencies) {
+    $excluded_dependencies_list = [];
+    foreach ($excluded_dependencies as $excluded_dependency) {
+      if (array_key_exists($excluded_dependency, $this->getLightningComponents())) {
+        $excluded_dependencies_list[] = $excluded_dependency;
+        if (in_array($excluded_dependency, $this->getTopLevelComponents())) {
+          // If its a top-level-component, add its subcomponents too.
+          $subcomponents = $this->getSubComponents(trim($excluded_dependency));
+          foreach ($subcomponents as $subcomponent) {
+            $excluded_dependencies_list[] = $subcomponent;
+          }
+        }
+      }
+    }
+    $this->excludedDependencies = array_merge($this->excludedDependencies, $excluded_dependencies_list);
+  }
+
+  public static function getLightningComponents() {
+    $appRoot = \Drupal::root();
+    $extensions = new ExtensionDiscovery($appRoot);
+    $extensions = $extensions->scan('module');
+
+    $lightning_extensions = self::array_filter_key($extensions, function($key) {
+      return strpos($key, 'lightning_') === 0;
```

Comment:
Maybe not the most reliable filter...I think we should filter by the path instead. Any extension located inside Lightning's modules/lightning_features directory (which we can divine by doing a separate scan for profiles) is a Lightning extension; everything else is not, even if it's prefixed with lightning_.